src/app/core/store/reducers/defendants.reducer.ts

```diff
@@ -65,48 +69,50 @@ export function reducer(
     }
 
     case fromDefendants.DefendantsActionTypes.LoadDefendantsSuccess: {
-      const theDefendants: Defendant[] = action.payload;
-      let theAssetId;
-
+      const payload: DefendantsPayload = action.payload;
```

Comment:
I don't see the need of creating a complex payload class since all the `AssetProperty` derived classed have the `getAssetId()` method.